plugins/services/src/js/components/forms/NetworkingFormSection.js

```diff
@@ -545,15 +545,25 @@ class NetworkingFormSection extends mixin(StoreMixin) {
   }
 
   getVirtualNetworks() {
+    const mesosContainer =
+      findNestedPropertyInObject(this.props.data, `container.type`) === "MESOS";
+
     return VirtualNetworksStore.getOverlays()
       .mapItems(overlay => {
         const name = overlay.getName();
 
         return {
+          enabled: overlay.info.enabled,
+          subnet6: overlay.getSubnet6(),
           text: name,
           value: `${CONTAINER}.${name}`
         };
       })
+      .filterItems(
+        virtualNetwork =>
+          virtualNetwork.enabled &&
+          (mesosContainer ? !virtualNetwork.subnet6 : true)
```

Comment:
We could add a small in-line function `isAvailableOnRuntime` just to clarify the intent:
```js
const isAvailableOnRuntime = (mesosContainer, {subnet6}) => !mesosContainer || !virtualNetwork.subnet6
```

How do you think? I'm afraid that complex logical expressions are hard to read.

Reply:
Done, I also added a comment.